include/flatbuffers/reflection.h

```diff
@@ -105,6 +105,20 @@ inline Table *GetFieldT(const Table &table,
   return table.GetPointer<Table *>(field.offset());
 }
 
+// Get a field, if you know it's a struct.
+inline const Struct *GetFieldStruct(const Table &table,
+                                    const reflection::Field &field) {
+  assert(field.type()->base_type() == reflection::Obj);
```

Comment:
You're right, it would be a bit clumsy to have to pass in the schema just for this check. Though without it, it is possible to call it on a field which is in fact a table, and then things will go badly wrong.

This is really my fault, because rather than the is_struct flag, it would have been better as a separate type.. but that is hard to change right now.

I guess at the minimum, stick a comment in front of the assert, like: "TODO: this does NOT check if the field is a table or struct, but we'd need access to the schema to check the is_struct flag."